src/mini_trainer/training_types.py

```diff
@@ -0,0 +1,68 @@
+"""
+Shared type definitions and dataclasses for mini_trainer.
+
+This module consolidates all common type definitions, enums, and dataclasses
+used across the mini_trainer package to avoid duplication and ensure consistency.
+"""
+
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Optional, Dict, Any
+
+
+class TrainingMode(str, Enum):
+    """Training mode determines the stopping criterion for training."""
+    EPOCH = "epoch"
+    STEP = "step"
+    TOKEN = "token"
+    INFINITE = "infinite"
+
+
+@dataclass
+class TorchrunArgs:
+    """Arguments for torchrun distributed training configuration."""
+    nnodes: int = 1
+    nproc_per_node: int = 1
+    node_rank: int = 0
+    rdzv_id: int = 123
+    rdzv_endpoint: str = "127.0.0.1"
+
+
+@dataclass
+class TrainingArgs:
+    """Complete training configuration arguments."""
+    # Required fields (no defaults)
+    model_name_or_path: str
+    data_path: str
+    batch_size: int
+    max_tokens_per_gpu: int
+    learning_rate: float
+    output_dir: str
+    
+    # Optional fields (with defaults)
+    num_warmup_steps: int = 0
+    lr_scheduler: str = "constant_with_warmup"
```

Comment:
Should this be changed to cosine default, given better performance?

Reply:
Ya lets change that